### PR TITLE
Changed all API calls that contain the word frame to tool

### PR DESCRIPTION
**Pull request by valentinptc**
I adjusted all APIs to rename frame to tool and reality to spatial.
@benptc can you test these APIs with all tools and the hardware interfaces?
Everything should be backward compatible.

### CODE REVIEW

**Comment by benptc** [comment]
OK, sounds good. I'll test these when I get a chance, before I approve it.

File: libraries/hardwareInterfaces.js [hunk 1]
```diff
@@ -440,44 +459,44 @@ exports.addNode = function (objectName, frameName, nodeName, type, position) {
...
 
-            if (!hardwareObjects.hasOwnProperty(objectName)) {
-                hardwareObjects[objectName] = new EmptyObject(objectName);
+            if (!hardwareObjects.hasOwnProperty(object)) {
+                hardwareObjects[object] = new EmptyObject(object);
             }
 
-            if (!hardwareObjects[objectName].frames.hasOwnProperty(frameUuid)) {
-                hardwareObjects[objectName].frames[frameUuid] = new EmptyFrame(frameName);
+            if (!hardwareObjects[object].rames.hasOwnProperty(frameUuid)) {
```
Comment (hobinjk-ptc) [inline libraries/hardwareInterfaces.js:479]: This is the first bug encountered when trying to run the server

File: libraries/hardwareInterfaces.js [hunk 2]
```diff
@@ -381,11 +400,11 @@ exports.setTool = function(objectName, frameName, tool, dirName){
...
-                objects[objectID].frames[frameUuid].tool = {addon: addonName, interface: interfaceName, tool: tool};
+                objects[objectID].frames[frameUuid].tool = {addon: addonName, interface: interfaceName, tool: newTool};
             }
         }
     }
-}
+};
 
 /**
  * @desc addIO() a new IO point to the specified RealityInterface
```
Comment (hobinjk-ptc) [inline libraries/hardwareInterfaces.js:410]: this jsdoc comment is now out of date

File: libraries/hardwareInterfaces.js [hunk 3]
```diff
@@ -344,14 +363,14 @@ exports.clearTool = function(objectName, frameName){
...
         }
     }
-}
+};
 
-exports.setTool = function(objectName, frameName, tool, dirName){
-  
-    var objectID = utilities.getObjectIdFromTarget(objectName, objectsPath);
-    console.log('set new tool: ', tool);
+exports.setTool = function (object, tool, newTool, dirName) {
```
Comment (hobinjk-ptc) [inline libraries/hardwareInterfaces.js:368]: the naming of tool and newTool here suggests that they're of the same data type/format but the code below appears to use them in different ways

Comment (hobinjk-ptc) [reply libraries/hardwareInterfaces.js:368]: this doesn't matter in the grand scheme of things

File: libraries/hardwareInterfaces.js (hunk 1 above)
Comment (hobinjk-ptc) [reply libraries/hardwareInterfaces.js]: (fixed in the commit I just pushed)

**Comment by hobinjk-ptc** [comment]
Merging this because it's going to break my other PR

**Pull request merged**